### PR TITLE
1580 document web modeler public api

### PR DESCRIPTION
**Pull request by pihme**
## What is the purpose of the change

Add documentation for Web Modeler Public API (beta)

## Are there related marketing activities

Not that I know of, but I am not in the look anyway

## When should this change go live?

preferably for next alpha release in January

## PR Checklist

- [X] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [X] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [X] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [X] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by pihme** [comment]
Long list of reviwers:
* @christian-konrad Please have a look at the disclaimer boxes in particular
* @jfriedenstab Please check for technical correctness
* @christinaausley I still have it in draft to clarify some details. I don't think the text will change much, though.

**Comment by pihme** [comment]
@akeller (with your SDK team and harmonization hat on) + @christian-konrad 

I would like to bring up the issue of naming. Internally we called it Public API because that was the working title in the PDP. This term is also used in the UI.

However, using Web Modeler Public API in the documentation is maybe not ideal for two reasons:
1. It looks unpleasant in the side bar
image
2. There already is a page Public API which defines public API in a certain way, e.g. the versioning scheme. () And this definition doesn't fit to what we intend to do.

So my question to you, Christian, is how important the public API part is for you, 

And my questions to you, Amara, are:
1. What is your recommendation from an SDK/API harmonization and alignment standpoint?
2. Are there any plans to overhaul the current definition of public API? The page already feels a bit dated

**Comment by akeller** [comment]
>There already is a page Public API which defines public API in a certain way, e.g. the versioning scheme. (https://docs.camunda.io/docs/apis-clients/public-api/) And this definition doesn't fit to what we intend to do.

Can you help me understand the definition you want to use or how it's different?

I have no current plans to update the Public API page, but that's because the SDK focus is still relatively new and we will dig into API and client docs next quarter. If you feel it's dated, that's enough for me to review the page and make a plan to update it.

**Comment by pihme** [comment]
> Can you help me understand the definition you want to use or how it's different?

TL;DR: The Public API page makes most sense when one assumes it is about the Zeebe API and our own Zeebe clients. And then the page was expanded but not updated properly.

Background:
* We now have a number of APIs which were developed by several teams and which follow different design paradigms.
* We have APIs in REST, GraphQL, and GRPC
* Some APIs work on an organization-wide level
    * Console API
    * Web Modeler API
* Other APIs work on a per-cluster level
    * Zeebe
    * Operate
    * Tasklist
    * Optimize
 * Identity API is not mentioned. I guess because so far we only use it internally. Wouldn't be surprised though, if we open it to customers for self-managed in the future
 * Only Zeebe API has clients, for the other APIs I am not aware of any clients
 * Only for Zeebe Clients the version schema MAJOR.MINOR:.PATCH is used.  One could debate whether it is true semantic versioning or product release roadmap versioning
 * For Operate and Web Modeler the versioning scheme is `/MAJOR/`. This is common for REST APIs that they look like `[baseURL]/v1/entity` and `[baseURL]/v2/entity`. And here the version of the API does not correspond to the version of the product
 * For GRPC there is no explicit versioning and it just takes the version from the product release (so it may increase minor and patch versions although nothing has changed in the API)
 * For Optimize and Console I am not an expert but from what I have seen it appears like there is no API versioning planned
 * For Tasklist / GraphQL seems to be unversioned as well. This appears to be a desired feature in GraphQL 

So looking at the Public API page, I see the following issues:
* Unclear which APIs are _officially supported_. The sentence mentions Tasklist and Zeebe explicitly, but the bullets below also mention Operate API. In my opinion all APIs documented are kinda officially supported. But since this is also a legal question, it is above my paygrade.
* The stuff about versioning only applies to Zeebe API really, and only to the clients we provide, not community clients
* The stuff about java package applies only to the Java client of Zeebe API
* The whole _Clients_  section in the documentation is all about _ Zeebe Clients_
* The other APIs have different versioning schemes

**Comment by christian-konrad** [comment]
@pihme @akeller Imo, we can completely ditch this page https://docs.camunda.io/docs/apis-clients/public-api/.
It does not add anything that is not already mentioned in the current documentation sections of the various APIs, only the versioning schema, and that should be merged to the Zeebe doc (as it applies only there). It is inconsistent and outdated (see the list of issues Peter mentioned), and probably confusing (it's headline looks like it covers a different, separate API in the list of APIs). The backwards compat notice should belong to the dedicated API doc pages.

https://docs.camunda.io/docs/apis-clients/working-with-apis-clients/ is also outdated (it says we offer Zeebe API + one additional public API + others? Very confusing).

@pihme the title for the doc section of the Web Modeler API should be just "**Web Modeler API (beta)**"

File: docs/self-managed/modeler/web-modeler/api.md
Comment (christian-konrad) [reply docs/self-managed/modeler/web-modeler/api.md:36]: We could also link to this page for further details:
https://docs.camunda.io/docs/self-managed/identity/user-guide/generating-m2m-tokens/

**Review by christian-konrad — CHANGES_REQUESTED** [review]
Small changes requested as follow-up on the "Public" naming discussion.

Overall, great and helpful documentation!

**Comment by pihme** [comment]
> @pihme @akeller Imo, we can completely ditch this page https://docs.camunda.io/docs/apis-clients/public-api/.

@akeller Do you agree? If so, should I remove the page now, or keep it in case you want to salvage some content from it?

File: docs/apis-clients/web-modeler-api/index.md
Comment (jfriedenstab) [inline docs/apis-clients/web-modeler-api/index.md:34]:
```suggestion
3. After creating the client, you can download a shell script to obtain a token.
```

File: docs/apis-clients/web-modeler-api/index.md
Comment (jfriedenstab) [inline docs/apis-clients/web-modeler-api/index.md:73]:
```suggestion
The API gives you access to the names as well as the Ids. For example, when requesting a file you will get the following information:
```

File: docs/apis-clients/web-modeler-api/index.md
Comment (jfriedenstab) [inline docs/apis-clients/web-modeler-api/index.md:75]:
```suggestion
- **simplePath** contains the human-readable path. This path may be ambiguous or may have ambiguous elements (e.g. folders) in it.
```

File: docs/apis-clients/web-modeler-api/index.md
Comment (jfriedenstab) [inline docs/apis-clients/web-modeler-api/index.md:76]:
```suggestion
- **canonicalPath** contains the unique path. It is a list of **PathElementDto** objects which contain the id and the name of the element.
```

Comment (jfriedenstab) [inline docs/apis-clients/web-modeler-api/index.md:76]: Maybe it would make sense to add an example here?

File: docs/apis-clients/web-modeler-api/index.md
Comment (jfriedenstab) [inline docs/apis-clients/web-modeler-api/index.md:78]: I think "ID" (instead of lowercase "id") is the correct spelling, so I'd suggest to use "ID" and "IDs" everywhere.

File: docs/self-managed/modeler/web-modeler/api.md
Comment (jfriedenstab) [inline docs/self-managed/modeler/web-modeler/api.md:20]:
```suggestion
A detailed API description is available as [OpenAPI](https://www.openapis.org/) specification at [http://localhost:8070/swagger-ui.html](http://localhost:8070/swagger-ui.html).
```

File: docs/self-managed/modeler/web-modeler/api.md
Comment (jfriedenstab) [inline docs/self-managed/modeler/web-modeler/api.md:28]: See my comments above.

File: docs/self-managed/modeler/web-modeler/api.md
Comment (jfriedenstab) [inline docs/self-managed/modeler/web-modeler/api.md:34]: A whitespace is missing here:
```suggestion
3. Obtain a token to access the REST API. You will need the `client_id` and `client_secret` from Identity application you created.
```

**Review by jfriedenstab — CHANGES_REQUESTED** [review]
Thanks, this will be helpful! :)

A few suggestions for improvement.

**Comment by akeller** [comment]
@pihme you can remove it, but you may also need to remove links to it on other pages. It should also redirect to the API overview. You may want to partner with @christinaausley on that. 

@felix-mueller is there any reason we need to have a page explicitly on Public API info? Or can we add this info to the individual APIs?

**Comment by pepopowitz** [comment]
Hi! This PR is currently in a state where merging will be blocked due to [a new `enforce-label` check](https://github.com/camunda/camunda-platform-docs/pull/1578) not having run. 

This is fixable! If you rebase on `main`, or merge `main`, this branch will get the changes it needs to run that check. 

Or if you'd prefer, @akeller or myself have the ability to check a box to merge without the check running. If/when you've got the reviews you need and you want us to merge for you, just let us know.

File: docs/apis-clients/web-modeler-api/index.md
Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md:39]: No. I am not sure refresh tokens are a thing for public APIs. In UIs it's different, because you have an open website or a session. So you have something where you can register a timer and refresh the token.

With M2M clients this is less common. Usually, you assume stateless and keep the token for as long as it is valid. Getting a new token vs. refreshing a token also makes little difference, technically. So I would not describe this, unless there is a need for it.

File: docs/apis-clients/web-modeler-api/index.md
Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md:11]: good catch

File: docs/apis-clients/web-modeler-api/index.md
Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md:32]: good catch

File: docs/apis-clients/web-modeler-api/index.md
Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md]: Maybe on the second iteration :clown_face:

File: docs/apis-clients/web-modeler-api/index.md
Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md]: @christinaausley Do you agree?

**Comment by pihme** [comment]
Doing rebase now

**Comment by pihme** [comment]
> It should also redirect to the API overview. You may want to partner with @christinaausley on that.

@christinaausley Happy to team up.

**Comment by pihme** [comment]
Related issue to rename it in the UI: 

* SaaS - https://github.com/camunda-cloud/console/issues/3827
* No such issue for Self managed yet. 

In Self managed we use the following names:
* Web Modeler API - refers to the internal API
* Web Modeler Public API (Beta) - refers to the public API

@christian-konrad  options here would be to leave it as is or to rename both to "Web Modeler Internal API" + "Web Modeler API (beta)". But not sure whether "Web Modeler API" is already established as a name with customers. Probably not, because it is in beta. But better safe than sorry.

/cc @Ben-Sheppard

**Comment by akeller** [comment]
@pihme @christian-konrad, we need to take the topic of deleting the APIs & Clients page, including the backward compatibility info, to the C8 Stakeholder meeting.

**Comment by christinaausley** [comment]
If https://docs.camunda.io/docs/apis-clients/public-api/ adds little value, I would recommend removing the page and implementing redirects in a separate, fresh PR -- ideally following merge of this PR. I find breaking larger PRs into smaller chunks assists with fixing any build errors and quick review. Happy to partner on this @pihme -- feel free to remove the page and I can hop in to add redirects, fix any broken links, and merge.

@christian-konrad To be clear, you recommend moving the information on the versioning schema to the Zeebe content? @pihme We could add this in the same aforementioned PR.

@christian-konrad Who can I partner with to ensure https://docs.camunda.io/docs/apis-clients/working-with-apis-clients/ is updated? We can launch a separate PR for this.

**Comment by christinaausley** [comment]
Regarding this PR @pihme, I'm about to push a commit with any technical adjustments. Are you still clarifying details here or can this PR be moved out of draft?

File: docs/apis-clients/web-modeler-api/index.md
Comment (christinaausley) [reply docs/apis-clients/web-modeler-api/index.md]: So this has been an interesting conversation since I started working with Camunda. Currently in our [Technical Writing Glossary](https://confluence.camunda.com/display/HAN/Technical+Writing+Glossary) [we use id](https://confluence.camunda.com/display/HAN/Technical+Writing+Glossary#:~:text=Id,For%20example%2C%20%60regionId%60.).

This is to mirror API responses/JSON values exactly as they are referenced in the text. Do you have strong thoughts either way here @jfriedenstab?

File: docs/apis-clients/web-modeler-api/index.md
Comment (christinaausley) [inline docs/apis-clients/web-modeler-api/index.md:31]: Are there specific steps to accomplish this?

File: docs/apis-clients/web-modeler-api/index.md
Comment (christinaausley) [inline docs/apis-clients/web-modeler-api/index.md:32]: How do you download this?

File: docs/apis-clients/web-modeler-api/index.md
Comment (christinaausley) [inline docs/apis-clients/web-modeler-api/index.md:33]: When you run the shell script?

File: docs/apis-clients/web-modeler-api/index.md
Comment (christinaausley) [inline docs/apis-clients/web-modeler-api/index.md:63]: We can likely remove this section altogether since we currently only have one topic. If this expands, I would suggest naming it something like "FAQ" instead. Therefore, ### Names versus ids can become a H2.

File: docs/apis-clients/web-modeler-api/index.md
Comment (christinaausley) [inline docs/apis-clients/web-modeler-api/index.md:69]: What do you mean by disambiguated here? I might use a simpler word.

**Review by christinaausley — CHANGES_REQUESTED** [review]
A few commits for technical adjustments -- let me know if you have any questions about these. I also added a few comments for clarity 👍

File: docs/apis-clients/web-modeler-api/index.md
Comment (jfriedenstab) [reply docs/apis-clients/web-modeler-api/index.md]: I don't have strong thoughts here :). In my opinion, especially the plural "IDs" is easier to read than "ids", but maybe that's just my impression. If we use "id" in the technical writing glossary, then let's just use it here, too. I also get your point regarding the JSON response.

By the way, this was my reference: https://dictionary.cambridge.org/dictionary/english/id

> "id": in psychoanalysis, the deepest part of the unconscious mind

> "ID": a set of numbers, letters, or symbols that is used to represent a piece of data or a process in a computer program

Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md]: I would keep it as is then, until there is an update on the technical writing glossary

**Comment by pihme** [comment]
> Regarding this PR @pihme, I'm about to push a commit with any technical adjustments. Are you still clarifying details here or can this PR be moved out of draft?

Can be moved out of draft. All clarifications are part of this comment thread and I think we are approaching the end.

File: docs/apis-clients/web-modeler-api/index.md
Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md:31]: Yes, but trivial ones. Once you create credentials you will see a UI with check boxes for Web Modeler. 
image

Screenshot might be helpful, but it is pretty much self explanatory, in my mind. Also since this is the first iteration of documentation on a beta feature, I think it can be a little rough around the edges.

Personally, I would prefer a page like this https://docs.camunda.io/docs/components/console/manage-clusters/manage-api-clients/ for Console API clients.

The page I linked is about managing API clients on a per-cluster basis. There is also a way to manage clients/credentials on an organization-wide basis. For this, there is currently no documentation. If there would be, I would link to it.

File: docs/apis-clients/web-modeler-api/index.md
Comment (pihme) [reply docs/apis-clients/web-modeler-api/index.md:32]: You click on the download button that appears after you clicked "Create" in the last UI. I admit that it is not clear if you only read this text, but if you read this text and do it at the same time, it should be pretty obvious. Also same comment applies to missing documentation on creating Console API credentials as above:

image

**Review by christian-konrad — APPROVED** [review]
Nice to read, thanks!

**Comment by pihme** [comment]
@christinaausley As this is my last work day before Christmas, this can be considered a hand over:
* Fully agree with additional PR you suggested https://github.com/camunda/camunda-platform-docs/pull/1581#issuecomment-1359848576
* Reminded Christian and Jan to complete their review
* Other than that I think this can be merged once you are happy with it. Feel free to make changes as you see fit.
* Merge could ideally be done before Christmas. If not, I would like to see it merged before the next alpha release.
* And we can always make it better later. This covers the first milestone. The next milestone will add more features and will likely need more documentation.

**Review by jfriedenstab — APPROVED** [review]
Looks good to me :)

**Comment by pihme** [comment]
> > In Self managed we use the following names:
> > 
> > * Web Modeler API - refers to the internal API
> > * Web Modeler Public API (Beta) - refers to the public API
> 
> > @christian-konrad options here would be to leave it as is or to rename both to "Web Modeler Internal API" + "Web Modeler API (beta)". But not sure whether "Web Modeler API" is already established as a name with customers. Probably not, because it is in beta. But better safe than sorry.
> 
> @pihme Renaming makes sense. The internal API should be marked clearly as internal, and I haven't seen the phrase used anywhere before.

Associated PR in identity: https://github.com/camunda-cloud/identity/pull/1448

**Comment by pihme** [comment]
> Hi! This PR is currently in a state where merging will be blocked due to [a new `enforce-label` check](https://github.com/camunda/camunda-platform-docs/pull/1578) not having run.
> 
> This is fixable! If you rebase on `main`, or merge `main`, this branch will get the changes it needs to run that check.
> 
> Or if you'd prefer, @akeller or myself have the ability to check a box to merge without the check running. If/when you've got the reviews you need and you want us to merge for you, just let us know.

@pepopowitz I did two rebase on main. THe check seems to be stuck somehow, and I don't know how to re-trigger it. I leave it to your capable hands

**Review by christinaausley — APPROVED** [review]

**Comment by christinaausley** [comment]
Just had to add a label here -- is this ready to merge?

**Comment by akeller** [comment]
I'm ok with merging this since it doesn't include removing the Public API page. I'll add a relevant issue for that work.

Just so I'm clear about the scope of this PR - it will only show up in the `/next/` docs. Is this intentional? It feels so because it's a Beta. We only do two docs updates a year that roll `/next/` into the default version automatically. Outside of that, you'll need to add changes to both `/docs/` and `/versioned_docs/version-8.1`.

**Comment by pihme** [comment]
> Just so I'm clear about the scope of this PR - it will only show up in the /next/ docs. Is this intentional? 

Yes, it is.

(I am not sure what the policy is for products that are on a continuous deployment basis, like Web Modeler. If you have a policy that it needs to be added to _latest_ and to _next_, then we would follow this policy. But intention-wise it is in _next_ with the prospect of removing the beta label before the minor release in April. Till then it's there for selected customers whom we can point personally to the pages in _next_)

**Comment by akeller** [comment]
@pihme I'm good with this and will merge.

I'll also add an item to our backlog to document (docs-on-docs) the process for continuously releasing components so it's clear where the documentation should go, particularly if its a beta.

**Pull request merged**
